rust/crd/src/supersetdb.rs

```diff
@@ -1,27 +1,77 @@
 use crate::{SupersetCluster, APP_NAME};
 use serde::{Deserialize, Serialize};
-use snafu::Snafu;
-use stackable_operator::builder::ObjectMetaBuilder;
-use stackable_operator::commons::product_image_selection::{ProductImage, ResolvedProductImage};
-use stackable_operator::k8s_openapi::apimachinery::pkg::apis::meta::v1::Time;
-use stackable_operator::k8s_openapi::chrono::Utc;
-use stackable_operator::kube::CustomResource;
-use stackable_operator::kube::ResourceExt;
-use stackable_operator::labels::{self, APP_VERSION_LABEL};
-use stackable_operator::schemars::{self, JsonSchema};
+use snafu::{ResultExt, Snafu};
+use stackable_operator::{
+    builder::ObjectMetaBuilder,
+    commons::product_image_selection::{ProductImage, ResolvedProductImage},
+    config::{
+        fragment::{self, Fragment, ValidationError},
+        merge::Merge,
+    },
+    k8s_openapi::{apimachinery::pkg::apis::meta::v1::Time, chrono::Utc},
+    kube::CustomResource,
+    kube::ResourceExt,
+    labels::{self, APP_VERSION_LABEL},
+    product_logging::{self, spec::Logging},
+    schemars::{self, JsonSchema},
+};
+use strum::{Display, EnumIter};
 
 #[derive(Snafu, Debug)]
 #[allow(clippy::enum_variant_names)]
 pub enum Error {
-    #[snafu(display("object is missing metadata to build owner reference"))]
-    ObjectMissingMetadataForOwnerRef {
-        source: stackable_operator::error::Error,
-    },
-    #[snafu(display("failed to retrieve superset version"))]
-    NoSupersetVersion,
+    #[snafu(display("fragment validation failure"))]
+    FragmentValidationFailure { source: ValidationError },
 }
 type Result<T, E = Error> = std::result::Result<T, E>;
 
+#[derive(
+    Clone,
+    Debug,
+    Deserialize,
+    Display,
+    Eq,
+    EnumIter,
+    JsonSchema,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    Serialize,
+)]
+#[serde(rename_all = "kebab-case")]
+#[strum(serialize_all = "kebab-case")]
+pub enum Container {
```

Comment:
Just nitpicking: Can we rename this to something like `InitDbContainer`?

Reply:
renamed in 2522efa80079f7ff22998f623dca26adc9245762